overlord/fdestate/fdestate.go

```diff
@@ -218,7 +219,12 @@ func initializeState(st *state.State) error {
 
 	// FIXME mount points will be different in recovery or factory-reset modes
 	// either inspect degraded.json, or use boot.HostUbuntuDataForMode()
-	dataUUID, dataErr := disksDMCryptUUIDFromMountPoint(dirs.SnapdStateDir(dirs.GlobalRootDir))
+	dataDir := dirs.GlobalRootDir
+	writable := filepath.Join(dirs.GlobalRootDir, "writable")
+	if osutil.FileExists(writable) {
+		dataDir = writable
+	}
```

Comment:
```suggestion
	if osutil.FileExists(writable) {
		// appears be an Ubuntu Core device, where /writable exists
		dataDir = writable
	}
```

Reply:
Thinking of it, I will test it is a mount point. Because someone could create `/writable` and easily break it.

Reply:
could we check whether `release` indicates we are on Core and so choose the right path?

Reply:
Sure. That might be just simpler.